packages/playwright-core/src/server/chromium/crPage.ts

```diff
@@ -369,6 +369,7 @@ export class CRPage implements PageDelegate {
   }
 
   async resetForReuse(): Promise<void> {
+    await this._mainFrameSession._client.send('Storage.clearDataForOrigin', { origin: '*', storageTypes: 'all' });
```

Comment:
Note: `file_systems` would work as well, but I feel like with that we are on the safer side.